src/olympia/addons/models.py

```diff
@@ -1005,15 +1003,23 @@ def attach_listed_authors(addons, addon_dict=None):
         if addon_dict is None:
             addon_dict = dict((a.id, a) for a in addons)
 
-        qs = (UserProfile.objects.no_cache()
+        qs = (UserProfile.objects
               .filter(addons__in=addons, addonuser__listed=True)
               .extra(select={'addon_id': 'addons_users.addon_id',
                              'position': 'addons_users.position'}))
         qs = sorted(qs, key=lambda u: (u.addon_id, u.position))
-        for addon_id, users in itertools.groupby(qs, key=lambda u: u.addon_id):
-            addon_dict[addon_id].listed_authors = list(users)
-        # FIXME: set listed_authors to empty list on addons without listed
-        # authors.
+
+        addons_with_authors = {
+            addon_id: sorted(users, key=lambda author: author.position)
+            for addon_id, users in itertools.groupby(
+                qs, key=lambda u: u.addon_id
+            )
+        }
+
+        for addon_id, addon in addon_dict.items():
+            if addon_id in addons_with_authors:
+                users = addons_with_authors[addon_id]
+                addon_dict[addon_id].listed_authors = users
```

Comment:
I'm still investigating a bit here, I'll file a follow-up issue to set this to an empty list. There are a few things that break in case I do so I left it unchanged in that regard. This isn't a huge performance-saver since there aren't that many add-ons without listed authors.